### ISSUE TITLE
preservatorError when connecting over Tailscale and using Tailscale SSH for  auth

### ISSUE THREAD

**Issue by sumitgouthaman** [xibbon/LaTerminal#13]
Fantastic app! Running into a minor issue when connecting using Tailscale SSH (fails on first try, works on immediate retry).

### My environment:
- Connecting to a Linux host from iPad/iPhone.
- Both devices are in the same Tailscale tailnet.
- Using [Tailscale SSH](https://tailscale.com/blog/tailscale-ssh/) feature.
  - With this feature, whenever I try to SSH into another device in my tailnet, the Tailscale app on my iPhone/iPad will show a puch notification.
  - Clicking on the push notification will switch me to the Tailscale app where I can re-login and approve the connection.
  - Once I approve, the SSH connection will continue.

### What I expect to happen:
- LaTerminal should connect successfully after I switch back to it from the Tailscale app.

### What happens:
- LaTerminal runs into "preservatorError" (see attached screenshots).
- However if I close the connection and try to connect again, it connects successfully.
  - The second time around Tailscale does not generate a notification to login, because the previous one was approved recently.
  - So it probably connects correctly because we don't have to switch apps in the middle of connecting.

### Steps to reproduce:
- Have an iPad and another server in the same Tailscale tailnet.
- Enable [Tailscale SSH](https://tailscale.com/blog/tailscale-ssh/).
- Activate Tailscale on the iPad.
- From the iPad try to SSH into the server.
- Click on the push notification that appears. Once the Tailscale app/website opens, login.
  - This will then drop you back to the tailscale console.
- Switch back to the LaTerminal app.

![LaTerminal_TailscaleSSH](https://user-images.githubusercontent.com/934190/191800779-f4f053c2-5037-4c5e-8ef2-fa1476b75836.PNG)
![LaTerminal_TailscaleSSH_2](https://user-images.githubusercontent.com/934190/191800799-2aa69c7f-1ab1-4a5e-a22c-563ba8f8b332.PNG)

**Comment by bradfitz** [comment]
I'm one of the Tailscale SSH developers (along with @maisem, @mihaip). We run a public SSH test server that simulates Tailscale SSH's auth type ("none") for SSH client authors to hit to test their SSH code against:

Observe OpenSSH works:

```
$ ssh -v -p  2222 188.166.70.128
....
debug1: Authentication succeeded (none).
```

Then compare your SSH client against that `188.166.70.128` (on port 2222).

**Comment by migueldeicaza** [comment]
I believe I have a workaround for a gap in the tailscale SSH server, would you like to try it out?

If so, you can try the build with Testflight here:

https://testflight.apple.com/join/cpPS8YnX

**Comment by sumitgouthaman** [comment]
I don't have my ipad with me to reproduce on the same device as comment #1. But on the iPhone with the testflight version, it does run into an error.
![IMG_0722](https://user-images.githubusercontent.com/934190/191873230-9e17d30f-ea42-461b-a970-24da85515585.jpg)

**Comment by migueldeicaza** [comment]
Thanks for sharing, I can reproduce this issue, sadly, it is not the fundamental problem, which is that it seems like the tailscale server sometimes does not accept SFTP connections.   I am trying to find out what is going on here.

**Comment by migueldeicaza** [comment]
Thanks to @bradfitz I know why you are getting the error above: what happens is that your tailscale is not authenticated, and that is why the SFTP connection hangs, it is waiting for the re-authentication to take place.   I could not see it, because I had some sort of do-not-disturb enabled (I was trying the iOS feature where it automatically guesses what notifications to let through, once I turned it off, I got the notification).

That said, I am working to detect the login request, and direct the user to that.
